examples/get.py

```
@@ -1,131 +1,93 @@
-                s.mount('https://', a)
-
-                try:
-                    logging.info(url)
-                    r = s.get(url, headers=headers,
-                              timeout=30, stream=True)
-
-                    if r.status_code != 200:
-                        logging.info(str(r.status_code) + ": " + str(r.text))
 
-                    else:
-                        self.resp = json.loads(r.content.decode("utf-8"))
+class NcmAPIv2(object):
+
+    def __init__(self):
+        self.resp_data = {'data': []}
+
+    @staticmethod
+    def establish_session():
+        with requests.Session() as session:
+            retries = Retry(total=5,  # Total number of retries to allow.
+                            backoff_factor=2,
+                            status_forcelist=[408,  # 408 Request Timeout
+                                              500,  # 500 Internal Server Error
+                                              502,  # 502 Bad Gateway
+                                              503,  # 503 Service Unavailable
+                                              504,  # 504 Gateway Timeout
+                                              ],
+                            )
+            a = HTTPAdapter(max_retries=retries)
+            session.mount('https://', a)
+
+        return session
+
+    def next_url(self):
+
+        for item in self.resp['data']:
+            self.resp_data['data'].append(item)
+
+        if args.page and self.resp['meta']['next']:
+            self.url = self.resp['meta']['next']
+            return self.url
+
+        if args.steps and self.resp['meta']['next']:
+            while args.steps != 0:
+                args.steps -= 1
+                self.url = self.resp['meta']['next']
+                return self.url
+
+    def get_data(self, get_url, json_output):
+
+        session = self.establish_session()
+
+        while get_url:
+
+            logging.info(get_url)
+
+            try:
```

I wouldn't bother with this try/except, just let it raise so the caller can have a better idea of what went wrong.